core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TlsSetupService.java

```diff
@@ -72,18 +71,21 @@ public void setupTls(CloudPlatform cloudPlatform, Stack stack, Map<String, Objec
             String tlsSetupScript = FileReaderUtils.readFileFromClasspath("init/tls-setup.sh");
             tlsSetupScript = tlsSetupScript.replace("$PUBLIC_IP", gateway.getPublicIp());
             final Session.Command tlsSetupCmd = tlsSetupSession.exec(tlsSetupScript);
-            tlsSetupCmd.join(TLS_SETUP_TIMEOUT, TimeUnit.SECONDS);
+            tlsSetupCmd.join(SETUP_TIMEOUT, TimeUnit.SECONDS);
             tlsSetupSession.close();
             final Session changeSshKeySession = ssh.startSession();
-            final Session.Command changeSshKeyCmd = changeSshKeySession.exec("echo '" + stack.getCredential().getPublicKey() + "' > ~/.ssh/authorized_keys");
-            changeSshKeyCmd.join(REMOVE_SSH_KEY_TIMEOUT, TimeUnit.SECONDS);
+            changeSshKeySession.allocateDefaultPTY();
+            String tmpSshRemoveScript = FileReaderUtils.readFileFromClasspath("init/remove-tmp-ssh.sh");
+            tmpSshRemoveScript = tmpSshRemoveScript.replace("$CLOUD_SSH_USER", cloudPlatformConnectors.get(stack.cloudPlatform()).getSSHUser());
+            final Session.Command tmpSshRemoveCmd = changeSshKeySession.exec(tmpSshRemoveScript);
```

Comment:
This script is basically 1 line. I think it is not necessary to store it in a separate file, so it won't be needed to read a file and replace the placeholder.